### PR TITLE
Feat response and request filters 0.16.x

### PR DESCRIPTION
**Pull request by PineappleIOnic**
## What does this PR do?

This PR implements the required response and request filters for 0.16.x to allow for backwards compatibility with prior SDK's

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?
Yes

### CODE REVIEW

File: tests/unit/Utopia/Request/Filters/V15Test.php [hunk 1]
```diff
@@ -0,0 +1,570 @@
...
+     * @dataProvider ordersProvider
+     * @dataProvider filtersProvider
+     */
+    public function testListDocuments(array $content, array $expected): void
+    {
+        $model = 'databases.listDocuments';
+
+        $result = $this->filter->parse($content, $model);
+
+        $this->assertEquals($expected, $result, 'fail');
```
Comment (TorstenDittmann) [inline tests/unit/Utopia/Request/Filters/V15Test.php:392]: is the 'fail' a leftover?

Comment (stnguyen90) [reply tests/unit/Utopia/Request/Filters/V15Test.php]: Yes 😅 . fixed

**Comment by stnguyen90** [comment]
What's left to do:

1. User
2. Variable
3. Webhooks
4. Probably additional usage responses
5. Various List models

**Comment by stnguyen90** [comment]
FYI, we'll need to handle time > duration after [this](https://github.com/appwrite/appwrite/pull/3805/commits/cb50220b6026445a6d1f6afd065b13cb97ae619d) is merged.

**Comment by stnguyen90** [comment]
> What's left to do:
> 
> 1. User
> 2. Variable
> 3. Webhooks
> 4. Probably additional usage responses
> 5. Various List models

Finished these. Should be ready.

**Review by stnguyen90 — APPROVED** [review]

**Review by PineappleIOnic — COMMENTED** [review]
lgtm

File: app/controllers/general.php [hunk 2]
```diff
@@ -194,6 +199,9 @@
                 case version_compare($responseFormat, '0.14.0', '<='):
                     Response::setFilter(new ResponseV14());
                     break;
+                case version_compare($responseFormat, '0.15.0', '<='):
```
Comment (christyjacob4) [inline app/controllers/general.php:202]:
```suggestion
                case version_compare($responseFormat, '0.15.3', '<='):
```

Comment (christyjacob4) [reply app/controllers/general.php:202]: Always use the highest patch version number 👍

File: src/Appwrite/Utopia/Response/Filters/V15.php [hunk 3]
```diff
@@ -0,0 +1,556 @@
...
+        if (isset($content['vars'])) {
+            $vars = [];
+            foreach ($content['vars'] as $i => $var) {
+                $vars[$var['key']] = $var['value'];
+            }
+            $content['vars'] = $vars;
+        }
+
+        $content = $this->parseDatetimeAttributes($content, ['$createdAt', '$updatedAt', 'scheduleNext', 'schedulePrevious']);
+        return $content;
```
Comment (christyjacob4) [inline src/Appwrite/Utopia/Response/Filters/V15.php:348]: we changed the `status` attribute which would previously be "enabled" or "disabled" to a boolean attribute  `enabled` which can be true or false 
https://github.com/appwrite/appwrite/pull/3798

File: src/Appwrite/Utopia/Response/Filters/V15.php [hunk 4]
```diff
@@ -0,0 +1,556 @@
...
+        $content = $this->parseDatetimeAttributes($content, ['$createdAt', '$updatedAt']);
+        return $content;
+    }
+
+    private function parseExecution($content)
+    {
+        unset($content['stdout']);
+        $content = $this->parsePermissions($content);
+        $content = $this->parseDatetimeAttributes($content, ['$createdAt', '$updatedAt', 'startTime', 'endTime']);
+        return $content;
```
Comment (christyjacob4) [inline src/Appwrite/Utopia/Response/Filters/V15.php:322]: The `time` attribute has been renamed to `duration`
https://github.com/appwrite/appwrite/pull/3801

File: app/controllers/general.php [hunk 5]
```diff
@@ -68,6 +70,9 @@
                 case version_compare($requestFormat, '0.14.0', '<'):
                     Request::setFilter(new RequestV14());
                     break;
+                case version_compare($requestFormat, '0.15.0', '<'):
```
Comment (christyjacob4) [inline app/controllers/general.php:73]:
```suggestion
                case version_compare($requestFormat, '0.15.3', '<'):
```

**Review by christyjacob4 — CHANGES_REQUESTED** [review]
Please address the comments

**Review by christyjacob4 — APPROVED** [review]
Looks good from me 👌

**Pull request merged**